pvlib/pvsystem.py

```diff
@@ -1683,6 +1675,128 @@ def dc_ohms_from_percent(self):
             self.strings)
 
 
+@dataclass
+class AbstractMount(ABC):
+    """
+    A base class for Mount classes to extend. It is not intended to be
+    instantiated directly.
+    """
+
+    @abstractmethod
+    def get_orientation(self, solar_zenith, solar_azimuth):
+        """
+        Determine module orientation.
+
+        Parameters
+        ----------
+        solar_zenith : numeric
+            Solar apparent zenith angle [degrees]
+        solar_azimuth : numeric
+            Solar azimuth angle [degrees]
+
+        Returns
+        -------
+        orientation : dict-like
+            A dict-like object with keys `'surface_tilt', 'surface_azimuth'`
```

Comment:
should we make this more explicit like 

A dict-like object with keys `'surface_tilt', 'surface_azimuth'` (typically a dict or pandas.DataFrame)